mapie/quantile_regression.py

```diff
@@ -0,0 +1,486 @@
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        if isinstance(estimator, Pipeline):
+            self._check_estimator(estimator[-1])
+            return estimator
+        else:
+            name_estimator = estimator.__class__.__name__
+            if name_estimator == "QuantileRegressor":
+                return estimator
+            else:
+                if name_estimator in self.quantile_estimator_params:
+                    param_estimator = estimator.get_params()
+                    loss_name, alpha_name = self.quantile_estimator_params[
+                        name_estimator
+                        ].values()
+                    if loss_name in param_estimator:
+                        if param_estimator[loss_name] != "quantile":
+                            raise ValueError(
+                                "You need to set the loss/objective argument"
+                                + " of your base model to ``quantile``."
+                            )
+                        else:
+                            if alpha_name in param_estimator:
+                                return estimator
+                            else:
+                                raise ValueError(
+                                    "The matching parameter alpha_name for"
+                                    + " estimator does not exist."
+                                )
+                    else:
+                        raise ValueError(
+                            "The matching parameter loss_name for"
+                            + " estimator does not exist."
+                        )
+                else:
+                    raise ValueError(
+                        "The base model does not seem to be accepted"
+                        + " by MapieQuantileRegressor. \n"
+                        "Give a base model among: \n"
+                        "``quantile_estimator_params.keys()``"
+                        "Or, add your base model to"
+                        + " ``quantile_estimator_params``"
+                    )
+
+    def _check_cv(
+        self,
+        cv: Optional[str] = None
+    ) -> str:
+        """
+        Check if cv argument is None or "split".
+
+        Parameters
+        ----------
+        cv : Optional[str], optional
+           cv to check, by default ``None``.
+
+        Returns
+        -------
+        str
+            cv itself or a default "split".
+
+        Raises
+        ------
+        ValueError
+            Raises an error if the cv is anything else but the method "split".
+            Only the split method has been implemented.
+        """
+        if cv is None:
+            return "split"
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method, only valid method is ``split``."
+            )
+
+    def fit(  # type: ignore
```

Comment:
Is the # type: ignore still needed ?

Reply:
Yes, otherwise you get this error:
"""
mapie/quantile_regression.py:311: error: Signature of "fit" incompatible with supertype "MapieRegressor"
mapie/quantile_regression.py:311: note:      Superclass:
mapie/quantile_regression.py:311: note:          def fit(self, X: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], y: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], sample_weight: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]], None] = ...) -> MapieRegressor
mapie/quantile_regression.py:311: note:      Subclass:
mapie/quantile_regression.py:311: note:          def fit(self, X: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], y: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], X_calib: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], y_calib: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], sample_weight: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]], None] = ...) -> MapieQuantileRegressor
"""

Reply:
Yes, indeed.